sklearn/metrics/cluster/supervised.py

```diff
@@ -610,9 +611,17 @@ def mutual_info_score(labels_true, labels_pred, contingency=None):
     normalized_mutual_info_score: Normalized Mutual Information
     """
     if contingency is None:
-        labels_true, labels_pred = check_clusterings(labels_true, labels_pred)
-        contingency = contingency_matrix(labels_true, labels_pred, sparse=True)
+        if labels_pred is None:
+            # labels_true is actually contingency
+            contingency = check_array(labels_true,
+                                      accept_sparse= ['csr', 'csc', 'coo'],
+                                      dtype=[int, np.int32, np.int64])
+
+        else:    
+            labels_true, labels_pred = check_clusterings(labels_true, labels_pred)
+            contingency = contingency_matrix(labels_true, labels_pred, sparse=True)
     else:
+        warnings.warn("Contingency is deprecated in 0.19 and will be removed in 0.21, to use contingency matrix pass only contingency matrix to scoring function", DeprecationWarning)
```

Comment:
Before you commit any changes, you should run all the necessary tests for it including pep8, pyflakes, etc. Helps in debugging locally. For more details on tests, refer the [contributor's guide](http://scikit-learn.org/stable/developers/contributing.html#contributing-pull-requests).